pkg/sdk/testint/tags_integration_test.go

```diff
@@ -281,3 +283,915 @@ func TestInt_TagsShowByID(t *testing.T) {
 		require.Equal(t, id2, e2.ID())
 	})
 }
+
+func TestInt_TagsAssociations(t *testing.T) {
+	client := testClient(t)
+	ctx := testContext(t)
+
+	awsBucketUrl := testenvs.GetOrSkipTest(t, testenvs.AwsExternalBucketUrl)
+	awsRoleARN := testenvs.GetOrSkipTest(t, testenvs.AwsExternalRoleArn)
+
+	tag, tagCleanup := testClientHelper().Tag.CreateTag(t)
+	t.Cleanup(tagCleanup)
+
+	tagValue := "abc"
+	tags := []sdk.TagAssociation{
+		{
+			Name:  tag.ID(),
+			Value: tagValue,
+		},
+	}
+	unsetTags := []sdk.ObjectIdentifier{
+		tag.ID(),
+	}
+
+	t.Run("TestInt_TagAssociationForAccount", func(t *testing.T) {
+		err := client.Accounts.Alter(ctx, &sdk.AlterAccountOptions{
+			SetTag: tags,
+		})
+		require.NoError(t, err)
+
+		returnedTagValue, err := client.SystemFunctions.GetTag(ctx, tag.ID(), testClientHelper().Ids.AccountIdentifierWithLocator(), sdk.ObjectTypeAccount)
+		require.NoError(t, err)
+		assert.Equal(t, tagValue, returnedTagValue)
+
+		err = client.Accounts.Alter(ctx, &sdk.AlterAccountOptions{
+			UnsetTag: unsetTags,
+		})
+		require.NoError(t, err)
+
+		_, err = client.SystemFunctions.GetTag(ctx, tag.ID(), testClientHelper().Ids.AccountIdentifierWithLocator(), sdk.ObjectTypeAccount)
+		require.ErrorContains(t, err, "sql: Scan error on column index 0, name \"TAG\": converting NULL to string is unsupported")
+	})
+
+	accountObjectTestCases := []struct {
+		name        string
+		objectType  sdk.ObjectType
+		setupObject func() sdk.AccountObjectIdentifier
+		setTags     func(sdk.AccountObjectIdentifier, []sdk.TagAssociation) error
+		unsetTags   func(sdk.AccountObjectIdentifier, []sdk.ObjectIdentifier) error
+	}{
+		{
+			name:       "ApplicationPackage",
+			objectType: sdk.ObjectTypeApplicationPackage,
+			setupObject: func() sdk.AccountObjectIdentifier {
+				appPackage, appPackageCleanup := testClientHelper().ApplicationPackage.CreateApplicationPackage(t)
+				t.Cleanup(appPackageCleanup)
+				return appPackage.ID()
+			},
+			setTags: func(id sdk.AccountObjectIdentifier, tags []sdk.TagAssociation) error {
```

Comment:
I am a bit confused here. We have `Tags#Set` method that is used in the `tags_association` resource and we have all the methods in specific interfaces. We moved all the tests here but we do not check `Tags#Set`. Can you explain in detail what's the plan after?

Reply:
`setTags` and `unsetTags` should be hidden behind another abstraction layer, it would be much easier to set these tests and also we would get closer to the method that can be easily used for tag association resource

Reply:
As we discussed, I've changed it to test both options for setting tags.